### ISSUE TITLE
Exception: Failed to parse time string (2018-03-11 14:06:40.-84412) at Carbon.php:408

### ISSUE THREAD

**Issue by nthachus** [briannesbitt/Carbon#1184]
This bug happen for v1.24.1 while calling `Carbon::now()` on PHP / Windows x64

While debugging, I found that the bug come from code line 398 `(microtime(true) * 1000000 % 1000000)`
So, I recommend to fix this bug by changing that code block to:
```php
$microTime = microtime(true);
$microTime = str_pad(strval(($microTime - (int)$microTime) * 1000000 % 1000000), 6, '0', STR_PAD_LEFT);
```

**Comment by Mod-joker** [comment]
Meeting the same bug.

```
Failed to parse time string (2018-03-12 11:27:51.-310156) at position 24 (1): Unexpected character
```

**Comment by alchimik** [comment]
PHP Version 5.6.23. Freebsd. 32bit

CODE:
```
$date = '2018-03-12 11:28:48';
$timestamp = strtotime($date);
dump($timestamp);

$xxx = str_pad( strval($timestamp * 1000000 % 1000000), 6, '0', STR_PAD_LEFT );
dump($xxx);
```

OUTPUT:
1520825328
"-677120"

related links:
- PHP - Large Integer mod calculation - https://stackoverflow.com/questions/759888/php-large-integer-mod-calculation
- modulus operation gives negative value in php - https://stackoverflow.com/questions/15740919/modulus-operation-gives-negative-value-in-php
- Note that operator % (modulus) works just with integers - http://php.net/manual/ru/language.operators.arithmetic.php

**Comment by Prabakaran-t** [comment]
Same Bug after updating 1.23.0 to 1.24.1 
`DateTime::__construct(): Failed to parse time string (2018-03-12 05:30:37.-118543) at position 24 (5): Unexpected character`

**Comment by kylekatarnls** [comment]
32bit problem identified.

Quick-fixed for 1.24.2

Discussed improvement on #1185
